core/crypto/client_tx.go

```diff
@@ -42,7 +46,11 @@ func (client *clientImpl) createDeployTx(chaincodeDeploymentSpec *obc.ChaincodeD
 	}
 
 	// Copy metadata from ChaincodeSpec
-	tx.Metadata = chaincodeDeploymentSpec.ChaincodeSpec.Metadata
+	tx.Metadata, err = getMetadata(chaincodeDeploymentSpec.GetChaincodeSpec(), tCert, attrs...)
```

Comment:
Why do we impose a structure on the metadata so tightened to the attributes? 

Moreover, this will be break compatibility with chaincodes that are already using metadata (asset_management.go) under the contract that the fabric does not impose any special formatting. 

If we want the fabric to impose a formatting on the metadata this should be in the direction of a separation between user metadata and transaction metadata. Then, the attributes metadata will be considered as transaction metadata, meaning that they are managed by the fabric. This approach will also offer retro-compatibility.